From 4c5abc3ca735c0cfcd04ea6a1c8674d3741f0d80 Mon Sep 17 00:00:00 2001
From: Marian Chereji <marian.chereji@freescale.com>
Date: Wed, 26 Feb 2014 10:42:32 +0200
Subject: [PATCH 565/987] dpa_offload: Add errors-to-discard optimization in
 dts files for T4240

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Adding the errors-to-discard option improves DPA offloading apps
performances. This optimization was accidentally omitted for the
T4240 platform.

Signed-off-by: Marian Chereji <marian.chereji@freescale.com>
Change-Id: Ib909b6bac08c6cc905aca4b4de2c8aeb4ee90a7a
Reviewed-on: http://git.am.freescale.net:8181/9142
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Aurelian Zanoschi <Aurelian.Zanoschi@freescale.com>
Reviewed-by: Jose Rivera <German.Rivera@freescale.com>
---
 drivers/staging/fsl_dpa_offload/dts/t4240si-chosen-offld.dtsi | 2 ++
 drivers/staging/fsl_dpa_offload/dts/t4240si-chosen-reass.dtsi | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/staging/fsl_dpa_offload/dts/t4240si-chosen-offld.dtsi b/drivers/staging/fsl_dpa_offload/dts/t4240si-chosen-offld.dtsi
index 352e7eb..ae36e38 100644
--- a/drivers/staging/fsl_dpa_offload/dts/t4240si-chosen-offld.dtsi
+++ b/drivers/staging/fsl_dpa_offload/dts/t4240si-chosen-offld.dtsi
@@ -10,6 +10,7 @@ chosen {
 			fman0_oh2-extd-args {
 				cell-index = <1>;
 				compatible = "fsl,fman-port-op-extended-args";
+				errors-to-discard = <0x00420000>;
 				/* Define Virtual storage profile */
 				/* <number of profiles, default profile id> */
 				vsp-window = <8 0>;
@@ -18,6 +19,7 @@ chosen {
 			fman0_oh3-extd-args {
 				cell-index = <2>;
 				compatible = "fsl,fman-port-op-extended-args";
+				errors-to-discard = <0x00420000>;
 				/* Define buffer layout parameters. Can be used
 				 * in fragmentation or header manip operations
 				 */
diff --git a/drivers/staging/fsl_dpa_offload/dts/t4240si-chosen-reass.dtsi b/drivers/staging/fsl_dpa_offload/dts/t4240si-chosen-reass.dtsi
index e52a554..85c0207 100644
--- a/drivers/staging/fsl_dpa_offload/dts/t4240si-chosen-reass.dtsi
+++ b/drivers/staging/fsl_dpa_offload/dts/t4240si-chosen-reass.dtsi
@@ -9,6 +9,7 @@ chosen {
 			fman0_rx1-extd-args {
 				cell-index = <1>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
+				errors-to-discard = <0x00020000>;
 				/* Define Virtual storage profile */
 				/* <number of profiles, default profile id> */
 				vsp-window = <8 0>;
@@ -16,6 +17,7 @@ chosen {
 			fman0_oh2-extd-args {
 				cell-index = <1>;
 				compatible = "fsl,fman-port-op-extended-args";
+				errors-to-discard = <0x00420000>;
 				/* Define buffer layout parameters. Can be used
 				 * in fragmentation or header manip operations
 				 */
-- 
1.9.1

